File: Leetcode/3. Longest Substring Without Repeating Characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int i,j,len,n,ans=0,k=0;
        n=s.length();
        int a[300]={0};
        len=0;
        for(i=0;i<n;i++)
        {
            len++;
            a[s[i]]++;
            if(a[s[i]]>1)
            {
                for(j=k;j<i;j++)
                {
                    if(s[j]==s[i])
                    break;
                    len--;
                    a[s[j]]--;
                }
                a[s[j]]--;
                len--;
                k=j+1;
            }
            ans=max(ans,len);
        }
        return ans;
    }
};

File: Leetcode/3. Longest Substring Without Repeating Characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int[] freq = new int[300];
        int n = s.length();
        if(n<2){
            return n;
        }
        int ans = 1;
        int left, right;
        left = right =0;
        int currLen = 1;
        freq[s.charAt(0)]++;
        for(int i=1;i<n;i++){
            char c = s.charAt(i);
            freq[c]++;
            if(freq[c] > 1){
                while(s.charAt(left)!=c){
                    freq[s.charAt(left)]--;
                    left++;
                }
                freq[c]--;
                left++;
            }
            right++;
            ans = Math.max(ans,right-left+1);
        }
        return ans;
    }
}
